uaclient/entitlements/base.py

```diff
@@ -145,45 +146,55 @@ def valid_service(self):
 
         return self._valid_service
 
-    def enable(self, *, silent_if_inapplicable: bool = False) -> bool:
+    # using Union instead of Optional here to signal that it may expand to
+    # support additional reason types in the future.
+    def enable(self) -> Tuple[bool, Union[None, CanEnableFailure]]:
         """Enable specific entitlement.
 
-        :param silent_if_inapplicable:
-            Don't emit any messages until after it has been determined that
-            this entitlement is applicable to the current machine.
-
-        @return: True on success, False otherwise.
+        @return: tuple of (success, optional reason)
+            (True, None) on success.
+            (False, reason) otherwise. reason is only non-None if it is a
+                populated CanEnableFailure reason. This may expand to
+                include other types of reasons in the future.
         """
         msg_ops = self.messaging.get("pre_enable", [])
         if not util.handle_message_operations(msg_ops):
-            return False
-        can_enable, _ = self.can_enable(silent=silent_if_inapplicable)
+            return False, None
+
+        can_enable, fail = self.can_enable()
         if not can_enable:
-            return False
+            if fail is None:
+                # this shouldn't happen, but if it does we shouldn't continue
+                return False, None
+            elif fail.reason == CanEnableFailureReason.INCOMPATIBLE_SERVICE:
+                # this is the only reason that won't necessarily stop us from
+                # enabling
+                handle_incompat_ret = self.handle_incompatible_services()
+                if not handle_incompat_ret:
+                    return False, fail
+            else:
+                # every other reason means we can't continue
+                if fail.message is not None:
+                    print(fail.message)
```

Comment:
Since we are now returning the `CanEnableFailure` here, I think we can move the `print` logic into the `cli` `action_enable` function